### DISCUSSION TITLE
[general] Warnings from custom format functions not ending up in `tar_meta()`

### DISCUSSION THREAD

**Discussion by Aariq** [ropensci/targets#1345] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I noticed recently that warnings coming from custom write functions in the `format` argument to `tar_target()` aren't in the data frame returned by `tar_meta()`.  Maybe that's intended to not confuse errors from `command` with errors from other places.  If however that sounds like a bug, I can work out a reprex.

**Comment by wlandau** [comment]
Added in b55b1726c014262657b1adfeabade718bd77ebf3

```r
library(targets)
list(
  tar_target(
    x,
    warning("run_warning"),
    format = tar_format(
      read = function(path) {
        readRDS(path)
      },
      write = function(object, path) {
        warning("storage_warning")
        saveRDS(object, path)
      }
    )
  )
)
```

```r
tar_make(reporter = "silent")
tar_meta(x, warnings)
#> # A tibble: 1 × 2
#>   name  warnings                    
#>   <chr> <chr>                       
#> 1 x     run_warning. storage_warning
```
